### PR TITLE
feat(nafnskirteini): new application

### PR DESCRIPTION
**Pull request by berglindoma13**
# ...

Attach a link to issue if relevant

## What

New application for ID cards, can be both with travelling privileges or without

Using already existing passport services that have been updated to receive a parameter for type of id, returning only ID cards and not passports. Calls already made to the service are unaffected since sending no parameter return the passports as it used to be. 

## Why

Specify why you need to achieve this

## Screenshots / Gifs
id1
id2
id3
id4
id5
id6
id7

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/shared/constants/src/lib/chargeItemCode.ts [hunk 1]
```diff
@@ -26,4 +26,8 @@ export enum ChargeItemCode {
   AOSH_STREET_REGISTRATION_SA116 = 'SA116', // Nýskráning
   AOSH_STREET_REGISTRATION_SA117 = 'SA117', // Rafræn forskráning
   HEALTHCARE_LICENSE_CERTIFICATE = 'L6102',
+  ID_CARD_REGULAR = 'AY105', // TODO get code
```
Comment (saevarma) [inline libs/shared/constants/src/lib/chargeItemCode.ts:29]: Should probably be done before merging?

**Review by saevarma — APPROVED** [review]
Core file LGTM

File: libs/clients/passports/src/lib/passports.config.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export const PassportsClientConfig = defineConfig<z.infer<typeof schema>>({
     return {
       xRoadServicePath: env.required(
         'XROAD_PASSPORT_LICENSE_PATH',
-        'IS-DEV/GOV/10001/SKRA-Protected/Forskraning-V1',
```
Comment (disaerna) [inline libs/clients/passports/src/lib/passports.config.ts:20]: Is this change relevant to all services using this client?

File: libs/island-ui/core/src/lib/RadioButton/RadioButton.tsx [hunk 3]
```diff
@@ -114,7 +114,7 @@ export const RadioButton = ({
           <Text as="span" fontWeight={checked ? 'semiBold' : 'light'}>
             {label}
           </Text>
-          {subLabel && large && (
```
Comment (disaerna) [inline libs/island-ui/core/src/lib/RadioButton/RadioButton.tsx:117]: Is sublabel only visible if the component is large as well?

File: libs/clients/passports/src/lib/passports.config.ts (hunk 2 above)
Comment (berglindoma13) [reply libs/clients/passports/src/lib/passports.config.ts:20]: Yes, they have updated their x-road for both passports and id-cards so this is relevant to all

File: libs/island-ui/core/src/lib/RadioButton/RadioButton.tsx (hunk 3 above)
Comment (berglindoma13) [reply libs/island-ui/core/src/lib/RadioButton/RadioButton.tsx:117]: I removed this so that we could have a sublabel on smaller radiobuttons as well, so it's just visible if there is any content in the sublabel parameter

File: libs/application/templates/id-card/src/fields/HasDisabilityLicenseMessage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    refetch()
+  }, [])
+
+  const { formatMessage } = useLocale()
+
+  return (
+    <Box style={{ fontSize: 14 }}>
+      {!loading ? (
+        <AlertMessage
+          message={
```
Comment (Toti91) [inline libs/application/templates/id-card/src/fields/HasDisabilityLicenseMessage/index.tsx:43]: Question: could this information be fetched in the initial data-providers step?

File: libs/application/templates/id-card/src/fields/SecondGuardianEmailAndPhone/index.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  | {
+      secondGuardianInformation?: {
+        email?: string
+        phoneNumber?: string
+      }
+      applicantInformation?: string
+    }
+  | undefined
+
+export const SecondGuardianEmailAndPhone: FC<
```
Comment (Toti91) [inline libs/application/templates/id-card/src/fields/SecondGuardianEmailAndPhone/index.tsx:24]: Why does this have to be a custom component?

File: libs/application/templates/id-card/src/forms/IdCardForm/ApplicantInformation/index.ts [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+            ) as NationalRegistryIndividual | undefined
+
+            const chosenApplicantNationalId = getValueViaPath(
+              formValue,
+              Routes.CHOSENAPPLICANTS,
+              '',
+            ) as string
+
+            return applicantIdentity?.nationalId === chosenApplicantNationalId
+          },
```
Comment (Toti91) [inline libs/application/templates/id-card/src/forms/IdCardForm/ApplicantInformation/index.ts:111]: This condition is repeated a lot here, I would extract to a utils file

File: libs/application/templates/id-card/src/forms/Review/Overview/index.ts [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+        }),
+        buildKeyValueField({
+          label: review.labels.deliveryOption,
+          colSpan: '6/12',
+          value: ({ answers }) => {
+            const priceChoice = getValueViaPath(
+              answers,
+              `${Routes.PRICELIST}.priceChoice`,
+            ) as string
+            console.log(priceChoice)
```
Comment (Toti91) [inline libs/application/templates/id-card/src/forms/Review/Overview/index.ts:233]: console.log

File: libs/application/templates/id-card/src/lib/IdCardTemplate.ts [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+
+const IdCardTemplate: ApplicationTemplate<
+  ApplicationContext,
+  ApplicationStateSchema<Events>,
+  Events
+> = {
+  type: ApplicationTypes.ID_CARD,
+  name: applicationMessage.name,
+  featureFlag: Features.idCardApplication,
+  dataSchema: IdCardSchema,
```
Comment (Toti91) [inline libs/application/templates/id-card/src/lib/IdCardTemplate.ts:49]: Missing `translationNamespaces`

File: libs/application/templates/id-card/src/lib/IdCardTemplate.ts [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+                ),
+              // read: {
+              //   externalData: ['submitPassportApplication'],
+              //   answers: ['passport', 'childsPersonalInfo'],
+              // },
+            },
+          ],
+          actionCard: {
+            pendingAction: {
+              title: coreHistoryMessages.applicationReceived,
```
Comment (Toti91) [inline libs/application/templates/id-card/src/lib/IdCardTemplate.ts:286]: Remove comments?

File: libs/application/templates/id-card/src/utils/formatText.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import { useLocale } from '@island.is/localization'
+
+export const GetFormattedText = (messageId: any) => {
```
Comment (Toti91) [inline libs/application/templates/id-card/src/utils/formatText.ts:3]: instead of any you could type this as `MessageDescriptor`

File: libs/application/ui-fields/src/lib/PaymentChargeOverviewFormField/PaymentChargeOverviewFormField.tsx [hunk 11]
```diff
@@ -28,6 +28,7 @@ export const PaymentChargeOverviewFormField: FC<
       chargeItemCode: string
     },
   ]
+  console.log('selectedChargeList', selectedChargeList)
```
Comment (Toti91) [inline libs/application/ui-fields/src/lib/PaymentChargeOverviewFormField/PaymentChargeOverviewFormField.tsx:31]: console.log

**Review by Toti91 — CHANGES_REQUESTED** [review]
A description would be nice and some screenshots :) 

Overall looks fine but there is a lot of code duplication, and lots of logic that could be extracted to a util function. I didn't comment on all of it, but I would just recommend trying to extract logic that is used in more than one place to a single function, for future maintainability

File: libs/application/templates/id-card/src/fields/HasDisabilityLicenseMessage/index.tsx (hunk 4 above)
Comment (berglindoma13) [reply libs/application/templates/id-card/src/fields/HasDisabilityLicenseMessage/index.tsx:43]: No it is not allowed

File: libs/application/templates/id-card/src/fields/SecondGuardianEmailAndPhone/index.tsx (hunk 5 above)
Comment (sigruntg) [reply libs/application/templates/id-card/src/fields/SecondGuardianEmailAndPhone/index.tsx:24]: Because if the user goes back, chooses another child with other second guardian, we will need have to clear out the email and phone for the new guardian, I could not do it on the page before, because it is required. If I would be able to remove this item from answers I would not have to do this.

File: libs/clients/passports/src/lib/passportsApi.service.ts [hunk 12]
```diff
@@ -113,12 +113,14 @@ export class PassportsService {
 
   async getIdentityDocument(
     auth: User,
+    type?: string | null,
```
Comment (thorkellmani) [inline libs/clients/passports/src/lib/passportsApi.service.ts:116]: Can this be a enum or a literal? It's not apparent what the possible types are, so if I were using this function I'd be forced to guess what the type _could_ be which isn't ideal

File: libs/clients/passports/src/lib/passportsApi.service.ts [hunk 13]
```diff
@@ -130,12 +132,14 @@ export class PassportsService {
 
   async getIdentityDocumentChildren(
     auth: User,
+    type?: string | null,
```
Comment (thorkellmani) [inline libs/clients/passports/src/lib/passportsApi.service.ts:135]: Same as above

File: libs/clients/passports/src/lib/passportsApi.service.ts [hunk 14]
```diff
@@ -220,9 +224,12 @@ export class PassportsService {
     }
   }
 
-  async getCurrentPassport(user: User): Promise<Passport> {
-    const userPassports = await this.getIdentityDocument(user)
-    const childPassports = await this.getIdentityDocumentChildren(user)
+  async getCurrentPassport(
+    user: User,
+    type?: string | null,
```
Comment (thorkellmani) [inline libs/clients/passports/src/lib/passportsApi.service.ts:229]: Same as above

**Review by thorkellmani — COMMENTED** [review]
One problem

File: libs/clients/passports/src/lib/passportsApi.service.ts (hunk 14 above)
Comment (berglindoma13) [reply libs/clients/passports/src/lib/passportsApi.service.ts]: done with all of those, only one available parameter at the moment, but I put it in a type

**Review by thorkellmani — APPROVED** [review]
Code-owned files (hugsmiðjan) LGTM 🌩️

**Review by disaerna — APPROVED** [review]
UI files look good 👍🏼

**Review by Toti91 — APPROVED** [review]

File: libs/application/templates/id-card/src/lib/messages/error.ts [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+    defaultMessage: 'Ekki tókst að sækja gögn',
+    description: 'Failed to fetch data error',
+  },
+  submitApplicationError: {
+    id: 'id.application:error.submitApplicationError',
+    defaultMessage: 'Villa kom upp við að skila inn umsókn',
+    description: 'Failed to submit application',
+  },
+  invalidValue: {
+    id: 'pp.application:error.invalidValue',
```
Comment (davidkef) [inline libs/application/templates/id-card/src/lib/messages/error.ts:36]: shouldn't this be id

File: libs/application/templates/id-card/src/lib/messages/error.ts [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    defaultMessage: 'Ógilt gildi.',
+    description: 'Error message when a value is invalid.',
+  },
+  invalidAgeTitle: {
+    id: 'pp.application:error.invalidAgeTitle',
```
Comment (davidkef) [inline libs/application/templates/id-card/src/lib/messages/error.ts:41]: shouldn't this be id

File: libs/application/templates/id-card/src/lib/messages/error.ts [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+    defaultMessage: 'Þú hefur ekki náð tilskyldum aldri fyrir þessa umsókn',
+    description: 'Error message when a value is invalid.',
+  },
+  invalidAgeDescription: {
+    id: 'pp.application:error.invalidAgeDescription',
```
Comment (davidkef) [inline libs/application/templates/id-card/src/lib/messages/error.ts:46]: shouldn't this be id

File: libs/application/templates/id-card/src/utils/isAvailableForApplication.ts [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+  today.setHours(0, 0, 0, 0)
+
+  // Calculate the date x months from today based on expiration limit
+  const xMonthsLater = new Date(today)
+  xMonthsLater.setMonth(xMonthsLater.getMonth() + EXPIRATION_LIMIT_MONTHS)
+
+  // Check if the input date is within the expiration limit
+  const withinExpirationDate = inputDate >= today && inputDate <= xMonthsLater
+
+  console.log('chosenType', idTypeChosen)
```
Comment (davidkef) [inline libs/application/templates/id-card/src/utils/isAvailableForApplication.ts:19]: remove console.logs

**Review by davidkef — APPROVED** [review]

File: libs/shared/form-fields/src/lib/RadioController/RadioController.spec.tsx [hunk 19]
```diff
@@ -61,17 +61,4 @@ describe('RadioController', () => {
...
-      { label: 'some checkbox', value: 'off', subLabel: 'nice sublabel' },
-      { label: 'another checkbox', value: 'on' },
-    ]
-    render(
-      <Wrapper>
-        <RadioController id="values" largeButtons={false} options={options} />
-      </Wrapper>,
-    )
-    expect(screen.queryByText('nice sublabel')).not.toBeInTheDocument()
-  })
```
Comment (Halli1414) [inline libs/shared/form-fields/src/lib/RadioController/RadioController.spec.tsx:76]: Why is this test being removed? 
If it is wrong is it possible to rather update it to be correct?

**Review by Halli1414 — APPROVED** [review]

**Review by sigruntg — APPROVED** [review]

**Review by svanaeinars — APPROVED** [review]

**Review by brynjarorng — APPROVED** [review]
LGTM
